src/main/java/ch/njol/skript/variables/Variables.java

```diff
@@ -237,22 +248,29 @@ public static boolean load() {
 						continue;
 					}
 
+					UnloadedStorage unloadedStorage = optional.get();
 					try {
-						@SuppressWarnings("unchecked")
-						Class<? extends VariablesStorage> storageClass = (Class<? extends VariablesStorage>) optional.get();
-						Constructor<?> constructor = storageClass.getDeclaredConstructor(String.class);
+						Class<? extends VariableStorage> storageClass = unloadedStorage.getStorageClass();
+						Constructor<?> constructor = storageClass.getDeclaredConstructor(SkriptAddon.class, String.class);
 						constructor.setAccessible(true);
-						variablesStorage = (VariablesStorage) constructor.newInstance(type);
+						variablesStorage = (VariableStorage) constructor.newInstance(unloadedStorage.getSource(), type);
 					} catch (InstantiationException | IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e) {
-						Skript.error("Failed to initalize database type '" + type + "'");
+						Skript.exception(e, "Failed to initalize database type '" + type + "' ensure constructors are properly created.");
 						successful = false;
 						continue;
 					}
 
+					if (variablesStorage instanceof SQLiteStorage) {
+						Skript.warning(
+								"Please be advised that SQL is single threaded and synchronous.\n" +
+								"Meaning you will be losing out on drastic performance.\n" +
```

Comment:
```suggestion
								"Meaning you will be losing out on drastic performance.\n" +
```
drastic performance what?
i think this should say `drastic performance improvements`